### ISSUE TITLE
NCPA 64bit

### ISSUE THREAD

**Issue by SuperGoodMike** [NagiosEnterprises/ncpa#144]
Hello Everyone,

Are you going to be releasing a NPCA 64 bit version with the 1.8 release?

**Comment by scot0357** [comment]
Yes I"m currently prepping a 64 bit version of Windows for this very purpose.

**Comment by jhomann** [comment]
This might actually be a lot harder than anticipated due to the fact the cx_Freeze only has a win32service base for Windows. Is there any particular reason you want a Windows 64bit version of NCPA? Unless you didn't mean the Windows version.

**Comment by celticguy42** [comment]
Any update on this?  We've been experimenting with removing Win32 support from our Hyper-V servers, and it works well, except that we use this agent.

**Comment by jhomann** [comment]
I was going to have this in the next version but it is not likely to happen right now. Unfortunately it seems that there is no windows 64 bit service available for cx_Freeze, which is what we use to bundle the entire product for win/linux/osx and I am not sure how to get around this. Will have to look into this more in the future and maybe ask the people behind cx_Freeze how this is possible.

**Comment by celticguy42** [comment]
With the release of Windows Server 2016, which includes Nano server, a 64 bit NCPA is more pressing now (Nano server does not have 32 bit support).  Have you had any luck with a 64 bit version of cx_Freeze?  Is there anything us non-developers can do to help get this done?

**Comment by jomann09** [comment]
If I can have some help figuring out a way to make a 64 bit only Windows environment then I will be able to test out cx_Freeze to find out whether or not it will be able to be built on it...  I was really hoping the new version would be out by NCPA 2.0.0 release time but it is looking like we are on our own. I will definitely *try* to build a 64 bit version for the 2.0.0 release when it comes up (should be very soon)

As for helping out, I will definitely post back here when I have a build to test - and you can definitely help with the testing portion. I'll keep this up to date with what I end up doing to try to get this working.

**Comment by celticguy42** [comment]
On 64 bit Windows 2012/7 or later, just open PowerShell and run:

remove-windowsfeature wow64-support

That will remove the wow64 (windows on Windows64) support that is used to run 32 bit apps.

**Comment by jomann09** [comment]
Thanks, I'll try that out and see if I can build one this weekend!

**Comment by jomann09** [comment]
Tested this out today... I'm having trouble now because NSIS doesn't have a 64-bit version... which it can install a 64-bit version of the program even as a 32-bit installer... but that makes it not able to work on a system that is 64-bit only. I'm going to try looking for alternatives too - might be the right time to change.

I'm going to focus on this after the 2.0.0 release comes out. Since I know we can currently build for everything else. Likely coming in a 2.1.0 release if I can get something working.

**Comment by rimvydukas** [comment]
Hi,
Almost 1 year has passed since last post. What are the news about 64 bit version of ncpa?

**Comment by jomann09** [comment]
We still can't build a 64bit version until we move away from NSIS since you can't install it on a 64bit only Windows machine. We can build a 64bit version and bundle it in with the 32bit version but it wouldn't give us anything new besides being a 64bit application.

I did take a look at WiX but haven't had time to try out making an installer for NCPA.

**Comment by ados-h** [comment]
I am a fan of NCPA.  Missing a 64bit Windows agent limits my capability for promoting it to my customers.  Really hope it could come to the world sooner.  Not sure if any assistance/hours would be needed from the crowd.

**Comment by jomann09** [comment]
I am currently working on a WiX build to replace our current NSIS installation so that we can build both a 32bit and 64bit installer. It definitely won't be finished until NCPA 3 through. If you know anything about WiX it'd be a help, but testing would also be great. I will be updating this issue in the future once more work has gone into it.

**Comment by ed-odonnell** [comment]
It has now been over two years since the last post. Are there any updates on this. I would like to have a 64bit NCPA and MSI installer.

I have some experience with WIX, and would like to help if I can.
